### PR TITLE
integrate layout into a new package

### PR DESCRIPTION
**Pull request by alex-bezek**
### Summary
Create a re-usable layout component to be utilized for our apps to configure and manage our navigation

### Additional Details
If you have anything else that you think may be relevant to this issue, list it here. Additional information can help us better understand your changes and speed up the review process.

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-consumer-layout/tests/nightwatch/memorial-herman/MemorialHerman.jsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import Layout from '../../../src/Layout';
+import AppShellExample from '../AppShellExample';
+
+const data = {
+  nav: {
+    quickLinks: [
```
Comment (zhongyn) [inline packages/terra-consumer-layout/tests/nightwatch/memorial-herman/MemorialHerman.jsx]: quicklinks are removed

File: packages/terra-consumer-layout/tests/nightwatch/memorial-herman/MemorialHerman.jsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        text: 'Log Daily Fitness',
+      },
+    ],
+    navItems: [
+      {
+        url: 'localhost:8080',
+        text: 'Home',
+      },
+      {
+        url: '?react_perf#/tests/nav-tests/default',
```
Comment (zhongyn) [inline packages/terra-consumer-layout/tests/nightwatch/memorial-herman/MemorialHerman.jsx]: if you wanna make it work with react roturer..

File: package.json [hunk 3]
```diff
@@ -89,14 +89,16 @@
     "react": "^15.6.1",
     "react-dom": "^15.6.1",
     "react-router": "^3.0.5",
+    "react-router-dom": "^4.2.2",
```
Comment (zhongyn) [inline package.json]: which component depends on this?

**Comment by zhongyn** [comment]
there is a dashed border in the main panel, is it expected?

screen shot 2017-09-05 at 1 48 33 pm

**Comment by alex-bezek** [comment]
@zhongyn yeah its just on the example page the content has a body with a border to show you where it is filling

File: packages/terra-consumer-nav/src/components/nav-items/NavItem.scss [hunk 4]
```diff
@@ -1,86 +1,73 @@
...
-    display: flex;
-  }
+  .link {
+    display: block;
+    text-decoration: none;
 
-  /* stylelint-disable selector-max-compound-selectors, max-nesting-depth */
-  .nav-item-wrapper {
-    padding-bottom: 20px;
+    &:active {}
```
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/components/nav-items/NavItem.scss]: unused class can be removed

File: packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss [hunk 5]
```diff
@@ -2,94 +2,71 @@
...
 :local {
   .profile {
-    background: none;
+    background: var(--terra-consumer-profile-link-background);
+    background-attachment: fixed;
+    background-size: cover;
     bottom: 0;
     position: fixed;
-    width: var(--terra-consumer--profile-width, $terra-consumer--nav-container-width);
+    width: var(--terra-consumer-profile-link-width, 300px);
```
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/components/user-profile/UserProfile.scss]: the rest of the nav is defaulted to 320, this should follow suit

**Comment by zhongyn** [comment]
nav background is cut off in IE11:

IE11:
screen shot 2017-09-05 at 2 20 06 pm

Chrome:
screen shot 2017-09-05 at 2 19 45 pm

File: packages/terra-consumer-layout/src/Layout.scss [hunk 6]
```diff
@@ -0,0 +1,89 @@
+@import './variables';
+
+:root {
+  --terra-consumer-body-background: url('https://wallpapercave.com/wp/5ynt7Gb.jpg');
+  --terra-consumer-nav-active-item-background: var(--terra-consumer-body-background);
+  --terra-consumer-nav-item-color: #fff;
+
+  --terra-consumer-nav-width: 300px;
```
Comment (Mahmud-Khan) [inline packages/terra-consumer-layout/src/Layout.scss]: Should the nav(300px) and nav container(320 in _variabless.scss ) be consistent ?

File: packages/terra-consumer-layout/src/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+$terra-tiny-breakpoint: 544px;
+$terra-medium-breakpoint: 992px;
+$terra-consumer--nav-container-width: 320px;
```
Comment (mhemesath) [inline packages/terra-consumer-layout/src/_variables.scss]: this shouldn't be a sass variable

File: packages/terra-consumer-layout/package.json [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
+    "react-router-dom": "^4.2.2",
+    "terra-base": "^1.0.0"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "prop-types": "^15.5.8",
+    "react-intl": "^2.3.0",
+    "terra-base": "^1.0.0",
```
Comment (mhemesath) [inline packages/terra-consumer-layout/package.json]: can you bump all of the terra dependencies to the latest version? They remove reset styles which mess portal up

**Comment by alex-bezek** [comment]
Iessues being pointed out are having corresponding sub-tasks on jira created to clean these up before releasing. we are fixing the build and then will address these issues as separate PR's directly into master rather than continuing to tack onto this massive branch

**Pull request merged**